src/lang.ts

```diff
@@ -73,14 +73,16 @@ function _mixin(kwArgs: MixinArgs): {} {
  * @param sources Any number of objects whose enumerable own properties will be copied to the target object
  * @return The modified target object
  */
-export function assign(target: {}, ...sources: {}[]): {} {
-	return _mixin({
-		deep: false,
-		inherited: false,
-		sources: sources,
-		target: target
-	});
-}
+export let assign = has('object-assign') ?
```

Comment:
An alternative way of doing this, instead of the `any` cast (that allows the interface to be preserved too) is the following:

```typescript
interface ObjectAssignConstructor extends ObjectConstructor {
    assign(target: {}; ...sources: {}[]): {};
}

function assign(target: {}; ...sources: {}[]): {} {
    return (typeof (<ObjectAssignConstructor> Object).assign === 'function') ? (<ObjectAssignConstructor> Object).assign.apply(null, arguments) : _mixin({
        deep: false,
        inherited: false,
        sources: sources,
        target: target
    });
```